spec.md

```diff
@@ -0,0 +1,238 @@
+- [Defining Supported Layer Types](#defining-supported-layer-types)
+- [Optional: Defining Config Schema](#optional-defining-config-schema)
+- [Optional: Artifact Publisher Manifest](#optional-defining-artifact-publisher-manifests)
+  - [Artifact Publisher Manifest Example](#artifact-publisher-manifest-type-example) as a [Well Known Type][def-well-known-types]
+
+## Scope
+
+The scope of Artifacts v1.0 is based on what exists in 1.0 of [manifest][image-manifest].
+
+This includes individual artifacts that use [OCI manifest][image-manifest] including [OCI Image][image-spec], [Helm][helm] and [Singularity][singularity].
+
+### Future Scope
+
+Future versions of artifacts will support collections of artifacts using [OCI Index][image-index].
+
+To support artifacts that represent a collection of other artifacts, a means to identify an index is a type of artifact will be required.
+
+## Defining OCI Artifact Types
+
+As registries become content addressable distribution points, tools that pull artifacts must know if they can operate on the artifact. Artifact types are equivalent to file extensions. When users open files, the host operating system typically launches the appropriate program. When users open a file, from within a program, the open dialog filters to the supported types. When search or security software scan the contents of a storage solution, the software must to know how to process the different types of content. When users view the contents of a storage solution, they see the textual and visual indications of the type. OCI Artifacts provides these core capabilities to [OCI distribution spec][distribution-spec] based registries.
+
+Authoring an OCI Artifacts involves the following steps:
+
+- Define a unique type
+- Define the format for other tools to operate upon the type
+- Define human elements, such as an icon and localized string to be displayed to users
+- Optionally, publish as a [well-known type][def-well-known-types] for registries to consume
+
+Defining a unique type, string and logo; artifacts can be represented as the following:
+
+|Icon|Artifact|`config.mediaType`|
+|-|-|-|
+|<img src="https://github.com/opencontainers/artwork/blob/master/oci/icon/color/oci-icon-color.png?raw=true" width=30x>|[OCI Image][image-spec]|`application/vnd.oci.image.config.v1+json`|
+|<img src="https://github.com/helm/helm-www/blob/master/themes/helm/static/img/apple-touch-icon.png?raw=true" width=30x>|[Helm Chart](https://helm.sh)|`application/vnd.cncf.helm.chart.config.v1+json`|
+|<img src="https://github.com/sylabs/singularity/blob/master/docs/logos/singularity_v3.png?raw=true" width=30x>|[Singularity][singularity], by [Sylabs][sylabs]|`application/vnd.sylabs.sif.config.v1+json`|
+
+## Defining a Unique Artifact Type
+
+Defining the unique type involves uniqueness for computer processing, and uniqueness for humans.
+
+For computer processing, artifacts are defined by setting `manifest.config.mediaType` to a globally unique value.
```

Comment:
> Would there be a process to 'register' a media type as officially used for a given artifact type?

[RFC 6838][1] already defines a universal registry of media types, and the artifact definitions defined in this spec could just piggy back on [the IANA registry][2] for any new media types folks want to declare for registries.  For pre-existing media types or media types that the media-type authors don't bother registering with the IANA, I think this will be objectively unenforceable ("both Alice and Bob can suggest artifact definitions for image/png"), but could be subjectively enforced by each implementing registry ("sorry Bob, I've already accepted Alice's artifact definition associated with image/png for *this* registry.  But Charlie didn't like Alice's artifact definition for his registry; maybe he'll like yours.").

[1]: https://tools.ietf.org/html/rfc6838
[2]: https://www.iana.org/assignments/media-types/media-types.xhtml

Reply:
Here's the previous PR that discussed the process: [optional-publishing-the-artifact-type](https://github.com/SteveLasker/artifacts/blob/artifact-authors-how-to/authoring-artifacts.md#optional-publishing-the-artifact-type). I need to re-work this as we moved to a spec, not just guidance. 
While registry operators can support anything they like, I am suggesting we post well-known artifact types here, for others to consume. That said, I'm open to ideas for better locations to store this, but it would need to store all the info under an [`artifactType.json`](https://github.com/SteveLasker/artifacts/tree/master/artifactTypes/vnd.oci.image.1). The premise is it's PR based, so it gets some review. 

As for how to get uniqueness, I'll read through RFC 6838 to see what I can glean.

Reply:
We're moving forward with the Iana registration process. I'm resolving this, just to keep focus on the open topics.